requirements.txt

```diff
@@ -1,4 +1,4 @@
 pytest
-pythonnet~=2.5.2
+pythonnet~=3.0.1
```

Comment:
I know we had to pin 2.5.2 prior to this PR, but do we want to pin to 3.0.1 now or just remove the version? @rtzoeller I seem to recall a conversation we had about pinning going forward but can't recall the decision.